docs/devel/kubemark-guide.md

```diff
@@ -0,0 +1,174 @@
+
+Currently single HollowNode uses 0.05 of a CPU core and ~60MB or memory, which taking into account default cluster addons and fluentD, allows running ~17.5 HollowNodes per
+core.
+
+#### Behind the scene details:
+
+Start-kubemark script does quite a lot of things:
+- Creates a master machine called hollow-cluster-master and PD for it (*uses gcloud, should be easy to do outside of GCE*)
+- Creates a firewall rule which opens port 443\* on the master machine (*uses gcloud, should be easy to do outside of GCE*)
+- Builds a Docker image for HollowNode from the current repository and pushes it to the Docker repository (*GCR for us, using scripts from `cluster/gce/util.sh` - it may get
+tricky outside of GCE*)
+- Generates certificates and kubeconfig files, writes a kubeconfig locally to `test/kubemark/kubeconfig.loc` and creates a Secret which stores kubeconfig for HollowKubelet/
+HollowProxy use (*used gcloud to transfer files to Master, should be easy to do outside of GCE*).
+- Creates a ReplicationController for HollowNodes and starts them up. (*will work exactly the same everywhere as long as MASTER_IP will be populated correctly, but you’ll need
+to update docker image address if you’re not using GCR and default image name*)
+- Waits until all HollowNodes are in the Running phase (*will work exactly the same everywhere*)
+
+<sub>\* Port 443 is a secured port on the master machine which is used for all external communication with the API server. In the last sentence *external* means all traffic
+comming from other machines, including all the Nodes, not only from outside of the cluster. Currently local components, i.e. ControllerManager and Scheduler talk with API server using insecure port 8080.</sub>
+
+### Running e2e tests on Kubemark cluster
+
+To run standard e2e test on your Kubemark cluster created in the previous step you execute `test/kubemark/run-e2e-tests.sh` script. It will configure ginkgo to
+use Kubemark cluster instead of something else and start a density test. This script should not need any changes to work on other cloud providers.
+
+If you want to run different test you just need to provide flags you want to be passed to `hack/ginkgo-e2e.sh` script which specified the regexp on tests that should be run,
+if you won't pass any flags to the script it will run Density '30 test.
+
+### Monitoring test execution and debugging problems
+
+Run-e2e-tests prints the same output on Kubemark as on ordinary e2e cluster, but if you need to dig deeper you need to learn how to debug HollowNodes and how Master
+machine (currently) differs from the ordinary one.
+
+If you need to debug master machine you can do similar things as you do on your ordinary master. The difference between Kubemark setup and ordinary setup is that in Kubemark
+etcd is run as a plain docker container, and all master components are run as normal processes. There’s no Kubelet overseeing them. Logs are stored in exactly the same place,
+i.e. `/var/logs/` directory.
```

Comment:
Can you explicitly mention here that this means that if any of those will crash/fail/stop etc., this means that they will never be restarted.